torch_xla/experimental/apply_layers.py

```diff
@@ -0,0 +1,126 @@
+from typing import Iterable, Mapping, Sequence, Dict, Tuple
+
+import torch
+import torch.nn as nn
+from torch.utils._pytree import tree_map
+from functorch.compile import default_partition
+
+import torch_xla
+from torch_xla.experimental.scan import scan
+
+
+def apply_layers(layers: Iterable[torch.nn.Module],
+                 input_data,
+                 partition_fn=default_partition):
+  """Applies each layer in `layers` to `input_data` sequentially.
```

Comment:
please follow https://github.com/pytorch/xla/blob/master/torch_xla/distributed/spmd/xla_sharding.py#L147-L158 for the comment format, this way it can be picked up by the docstring later when we want to make it user facing.

Reply:
after render https://pytorch.org/xla/master/learn/api-guide.html#pytorch-xla-api

Reply:
done